src/warehouse/index.js

```diff
@@ -987,11 +987,23 @@ function processWarehouseMessage(message, options) {
 
       const usersEvent = { ...commonProps };
       const usersColumnTypes = {};
+      let userColumnMappingRules = whUserColumnMappingRules;
+      if (!isDataLakeProvider(options.provider)) {
```

Comment:
We are populating it for all non Datalake destinations because there are few destinations like CLICKHOUSE, RS, POSTGRES where load files are being used in certain configurations to load the data. 
1. [UploadJobUsersAppendRecordsUsingUsersLoadFiles](https://github.com/rudderlabs/rudder-server/blob/336b876f2d5dc334359d99593baa2b9295769541/warehouse/integrations/testhelper/records.go#L239C1-L240C1)
2. [UploadJobUsersRecordsUsingUsersLoadFilesForClickhouse](https://github.com/rudderlabs/rudder-server/blob/336b876f2d5dc334359d99593baa2b9295769541/warehouse/integrations/testhelper/records.go#L262)
Once these changes are merged we would need to update these test cases as well.